### ISSUE TITLE
Fill-paragraph should ignore headers

### ISSUE THREAD

**Issue by mok0** [jrblevin/markdown-mode#159]
I installed markdown-mode a couple of days ago from the  MELPA Stable, I am not sure what version I have.

When I use M-q to do fill-paragraph on a paragraph within markdown-mode, the text and headers are folded into the paragraph, so this:

```
# This is a header
Sed egestas molestie elit. Mauris urna mi, scelerisque vitae, ultrices vel, euismod vel, eros. Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Pellentesque dictum felis a nisi.
```

becomes this:

```
# This is a header Sed egestas molestie elit. Mauris urna mi,
scelerisque vitae, ultrices vel, euismod vel, eros. Lorem ipsum dolor
sit amet, consectetuer adipiscing elit. Pellentesque dictum felis a
nisi.
```

If there is a second level header, it becomes even worse:

```
# This is a header # Another header Sed egestas molestie elit. Mauris
#urna mi, scelerisque vitae, ultrices vel, euismod vel, eros. Lorem
#ipsum dolor sit amet, consectetuer adipiscing elit. Pellentesque
#dictum felis a nisi.
```

I think this is not the right behaviour, headers should not be filled. Org-mode does it correctly.

**Comment by vyp** [comment]
> I am not sure what version I have.

`M-x describe-package` :wink:

**Comment by mok0** [comment]
Thanks, @vyp! This is the output from M-x describe-package:
```
    Status: Installed in `~/.emacs.d/elpa/markdown-mode-2.1/' (unsigned).
    Archive: n/a
    Version: 2.1
   Requires: cl-lib-0.5
    Summary: Emacs Major mode for Markdown-formatted text files
   Homepage: http://jblevins.org/projects/markdown-mode/
   Keywords: [markdown] [github flavored markdown] [itex]
    Other versions: 2.1 (melpa-stable).
```

**Comment by syohex** [comment]
How about #162 ?

**Comment by mok0** [comment]
Wow, a one line fix... it works perfectly, thanks a lot!

**Comment by syohex** [comment]
#162 was merged.

**Comment by jrblevin** [comment]
Many thanks for the report, @mok0, and for the patch, @syohex.
